python/tvm/relay/frontend/tensorflow.py

```diff
@@ -942,7 +942,10 @@ def _impl(inputs, attr, params, mod):
         )
 
         if sparse_lhs:
-            data = _op.transpose(data)
+            if attr.get("adjoint_a"):
```

Comment:
This logic is very confusing. I've looked over it a couple of time and I'm still not sure if it is correct. How about you document all the cases like this:
```
A B -> (B.T A.T).T -> transpose(sparse_dense(transpose(B), A, sparse_lhs=false))
A B.T -> ??
A.T B -> ??
A.T B.T -> ??
```
And then you can make a series of if statements that match these cases.

Reply:
Yes i agree the logic is not straight forward, however it is to match the operation towards Topi Op.
I think matching series of If may not be good as it will branch out complete code if we want to tag with each case.
But i can try putting some comment over the current code, so that it will help to understand logic better. Please let me know if you think otherwise.

Reply:
@tkonolige : I have added few comments, would you please take a look. Thanks!

Reply:
Given the comment has 4 cases, I think it is most intuitive if the if-cases follow the same structure.
```python
if adjoint_a and adjoint_b:
  B = transpose(B)
  sparse_lhs = True
elif adjoint_a and not adjoint_b:
  ...
```

Reply:
Done!